all-sdk/external-client-configuration.md

```diff
@@ -0,0 +1,551 @@
+      * Not being clear this is for mTLS client use is confusing to users. And we'd have to change the path one anyways
+        because the CLI is not suffixed with path.
+  * `TEMPORAL_TLS_X` for all other values - ones that don't match need to be deprecated and replaced with newer forms.
+  * `TEMPORAL_CODEC_ENDPOINT` - works just fine.
+  * `TEMPORAL_CODEC_AUTH` - works just fine.
+  * 💭 Why deprecate some instead of making all our clients support these variables that exist in CLI?
+    * It is more sane/reasonable to have a simple to understand config-field-to-env-var format than it is to have these
+      special environment variables be inconsistent outliers.
+    * For non-CLI uses (i.e. SDKs), it is unreasonable to burden them with past CLI choices.
+* For gRPC metadata (i.e. HTTP headers), the environment variable format is `TEMPORAL_GRPC_META_<name>`.
+  * `<name>` is canonicalized into HTTP header format (gRPC libraries do this for you).
+  * Like all HTTP headers, comma-delimited values are supported for multi-headers.
+  * This does require that environment variable lists be scanned for prefixes. This is deemed an acceptable tradeoff.
+  * 💭 Why not `TEMPORAL_GRPC_META` as a single var that accepts some kind of structured format?
+    * There isn't really a good structured format. Comma-delimited key=value would require users to escape commas in the
+      value (common in HTTP values), there's not a good delimiter for other-delimited key=value, and requiring a JSON
+      object as the env var value is a bit hard to use.
+  * 💭 Why not `TEMPORAL_GRPC_META_<index>` as `key: value`?
+    * Scanning has to happen anyways.
+    * `TEMPORAL_GRPC_META_AUTHORIZATION` as `Bearer my-token` is cleaner than `TEMPORAL_GRPC_META_0` as
+      `Authorization: Bearer my-token`.
+    * Users shouldn't have to keep up with indexes.
+  * 💭 Why not `TEMPORAL_GRPC_META` as a multiline set of headers?
+    * This can be added later if wanted.
+    * People want to be able to set individual headers.
+    * If we use traditional header format, requires HTTP header parsing which is not as trivial as it may seem (and
+      parsers may not be present/accessible in every standard library). Granted we could accept our own multiline format
+      or only support a simple subset of the header format.
+* There are some special environment variables respected by clients:
+  * `TEMPORAL_PROFILE` - the name of the profile to load if one is not provided at load time. The default is  `default`.
+  * `TEMPORAL_CONFIG_FILE` - the path to the config file. The default is `~/.config/temporalio/temporal-client.toml`.
+
+### Loading Configuration
+
+Configuration is loaded in the following manner:
+
+* Try to load profile from configuration file
+  * Only if file loading is enabled (which it is by default)
+  * If user provides specific config file, use that, otherwise use the default location
+  * If user specifies the profile, use that, otherwise use the default
```

Comment:
Do we want to add a step in here somewhere with the config-file-specific absolute path validation?